Data/ByteString/Lazy.hs

```diff
@@ -696,6 +726,43 @@ drop i cs0 = drop' i cs0
             then Chunk (S.drop (fromIntegral n) c) cs
             else drop' (n - fromIntegral (S.length c)) cs
 
+-- | /O(n)/ @'dropEnd' n xs@ is equivalent to @'take' ('length' xs - n) xs@.
+-- Drops @n@ elements from end of bytestring.
+--
+-- >>> dropEnd 3 "abcdefg"
+-- "abcd"
+-- >>> dropEnd 0 "abcdefg"
+-- "abcdefg"
+-- >>> dropEnd 4 "abc"
+-- ""
+--
+-- @since 0.11.2.0
+dropEnd :: Int64 -> ByteString -> ByteString
+dropEnd i p | i <= 0 = p
+dropEnd i p = go [] [] 0 0 p
+  where go hss tss acc h (Chunk c cs)
+            | h >= acc - i  = go hss (c : tss) (acc + len c) h cs
+            | otherwise =
+              let (output, hss', tss', acc') = getOutput [] hss (c : tss) ( acc + len c)
+                in L.foldl (flip chunk) (go hss' tss' acc' (hLen hss') cs) output
+        go hss tss _ _ Empty = dropChunks (tss ++ L.reverse hss) (fromIntegral i)
+
+        len c = fromIntegral (S.length c)
+        hLen cs = maybe 0 len (listToMaybe cs)
+
+        getOutput out [] [] acc = (out, [], [], acc)
+        getOutput out [] bss acc = getOutput out (L.reverse bss) [] acc
+        getOutput out (x:xs) bss acc =
+            if len x <= acc - i - len x
```

Comment:
Looks suspicious, there are `len x` on both sides. Wouldn't `len x <= acc - i` be enough?

Reply:
Yes, its redundant